dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/WebContainer.java

```diff
@@ -606,6 +608,19 @@ public void removeWebApplication(DeployedModule deployedModule) throws Exception
      * @return WebContainer
      */
     public static WebContainer getWebContainer() {
+        WebContainer selfInstance = self.get();
+        if (selfInstance != null) {
+            return selfInstance;
+        }
+        CountDownLatch currentLatch = selfInit;
+        try {
+            currentLatch.await(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            // auto-FFDC
+            Thread.currentThread().interrupt();
```

Comment:
Are you meaning to re-interrupt the thread or the remove the interrupt flag from the thread?  If the latter you will want to used the interrupted method instead of the interrupt method.

Reply:
I did intetend to re-interrupt the thread here.